zebrad/src/components/sync/downloads.rs

```diff
@@ -471,17 +482,11 @@ where
 
                     metrics::counter!("sync.max.height.limit.paused.count", 1);
                 } else if block_height <= lookahead_reset_height && past_lookahead_limit_receiver.cloned_watch_data() {
-                    // Try to reset the watched value to false, since we're well under the limit.
-                    match past_lookahead_limit_sender.try_lock() {
-                        Ok(watch_sender_guard) => {
-                            // If Zebra is shutting down, ignore the send error.
-                            let _ = watch_sender_guard.send(true);
-                            metrics::counter!("sync.max.height.limit.reset.count", 1);
-                        },
-                        Err(TryLockError::Poisoned(_)) => panic!("thread panicked while holding the past_lookahead_limit_sender mutex guard"),
-                        // We'll try allowing new downloads when we get the next block
-                        Err(TryLockError::WouldBlock) => {}
-                    }
+                    // Reset the watched value to false, since we're well under the limit.
+                    // We need to block here, because if we don't the syncer can hang.
+
+                    // But if Zebra is shutting down, ignore the send error.
+                    let _ = past_lookahead_limit_sender.lock().expect("thread panicked while holding the past_lookahead_limit_sender mutex guard").send(false);
```

Comment:
Oops, I accidentally deleted this metric:

```suggestion
                    let _ = past_lookahead_limit_sender.lock().expect("thread panicked while holding the past_lookahead_limit_sender mutex guard").send(false);
                    metrics::counter!("sync.max.height.limit.reset.count", 1);
```